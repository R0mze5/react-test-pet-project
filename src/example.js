import { timeout } from 'q';

export const sum = (a, b) => a + b;
export const immutablePush = (arr, val) => {
  const newArr = arr.concat();
  newArr.push(val);
  return newArr;
};
export const intervalPromise = (fn, timeout) => {
  return new Promise(resolve => setTimeout(() => resolve(fn()), timeout));
};
