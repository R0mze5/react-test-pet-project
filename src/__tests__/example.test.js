import { sum, immutablePush, intervalPromise } from '../example';

describe('sum testing', () => {
  beforeEach(() => {});
  afterEach(() => {});
  beforeAll(() => {});
  afterAll(() => {});
  test('add 1 + 2 equal 3', () => {
    expect(sum(1, 2)).toBe(3);
  });

  // it == test
  it('add 1 + 2 equal 3', () => {
    expect(sum(1, 2)).not.toBe(4);
  });

  // snaphot'ы делаются при первом вызове, а при следующих проверяется новый результат вызова функции со snapshot
  it('add 1 + 2 equal 3', () => {
    expect(sum(1, 2)).toMatchSnapshot();
  });
});

// toBe - равно
// toEqual - глубокое сравнение
it('mutation in Array', () => {
  const arr = [1, 2, 3];

  expect(immutablePush(arr, 4)).toEqual([...arr, 4]);
});

describe('проверка асихронного кода', () => {
  it('promiseTimeout вернет Promise', () => {
    expect(intervalPromise(() => 'test', 1000)).toBeInstanceOf(Promise);
  });
  it('promiseTimeout вернет Promise, который исполнится по timeout', done => {
    expect(intervalPromise(done, 1000)).toBeInstanceOf(Promise);
  });
  it('вернет верный результат из Promise', () => {
    expect(intervalPromise(() => 'test string', 1000)).resolves.toEqual('test string');
  });
});
