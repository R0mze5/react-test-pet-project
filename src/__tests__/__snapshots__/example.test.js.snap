// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sum testing add 1 + 2 equal 3 1`] = `3`;
